environs/config/config.go

```diff
@@ -1335,13 +1343,39 @@ func (c *Config) NumProvisionWorkers() int {
 	return value
 }
 
+// validateNumProvisionWorkers ensures the number cannot be set to
+// more than 100.
+// TODO: (hml) 26-Feb-2024
+// Once we can better link the controller config and the model config,
+// allow the max value to be set in the controller config.
+func (c *Config) validateNumProvisionWorkers() error {
+	value, ok := c.defined[NumProvisionWorkersKey].(int)
+	if ok && value > 100 {
```

Comment:
we should pull out an embedded constant into a constant block near the top of the file rather than having a magic value inside the code.
```
const maxNumProvisionWorkers = 100
```
or something to that effect.